sdk/python/kfp/cli/cli_test.py

```diff
@@ -166,6 +166,82 @@ def test_deprecation_warning(self):
                       res.stdout.decode('utf-8'))
 
 
+class TestKfpDslCompile(unittest.TestCase):
+
+    def invoke(self, args):
+        starting_args = ['dsl', 'compile']
+        args = starting_args + args
+        runner = testing.CliRunner()
+        return runner.invoke(
+            cli=cli.cli, args=args, catch_exceptions=False, obj={})
+
+    def test_compile_with_caching_flag_enabled(self):
+        with tempfile.NamedTemporaryFile(suffix='.py') as temp_pipeline:
+            # Write the pipeline function to a temporary file
+            temp_pipeline.write(b"""
```

Comment:
Why don't we just have a static file we can reference, rahter than trying to dynamically write and use the file via temp files? It would be easier and cleaner to manage, especially since we are writing essentially the same file 3 separate times.  

That said, if there is a technical reason for using tempfiles, since it appears the pipelines are the same across all 3 `test_compile_ functions, we should extract this 'tiny-pipeline' into a variable and simply reference it here instead

Reply:
Done, created a static file that gets referenced each time.